packages/foundry/contracts/Witch.sol

```diff
@@ -225,13 +227,37 @@ contract Witch is AccessControl {
         DataTypes.Balances memory balances,
         DataTypes.Debt memory debt
     ) internal view returns (DataTypes.Auction memory) {
-        // We store the proportion of the vault to auction, which is the whole vault if the debt would be below dust.
-        DataTypes.Line storage line = lines[vault.ilkId][series.baseId];
-        uint128 art = uint256(balances.art).wmul(line.proportion).u128();
-        if (art < debt.min * (10**debt.dec)) art = balances.art;
-        uint128 ink = (art == balances.art)
-            ? balances.ink
-            : uint256(balances.ink).wmul(line.proportion).u128();
+        // We try to partially liquidate the vault if possible.
```

Comment:
Here, please use `uint256` for all intermediate calculations (`min`, `art`, `ink`), and cast down to `uint128` only when calling a function, creating a struct, emitting an event or returning a value.

Reply:
ok